common/src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/info/MaterialIconType.java

```diff
@@ -130,10 +131,12 @@ public ResourceLocation getBlockTexturePath(@Nonnull MaterialIconSet materialIco
 
         MaterialIconSet iconSet = materialIconSet;
         //noinspection ConstantConditions
-        if (!iconSet.isRootIconset && Platform.isClient() && Minecraft.getInstance() != null && Minecraft.getInstance().getResourceManager() != null) { // check minecraft for null for CI environments
+        if (!Platform.isClient() || Minecraft.getInstance() == null || Minecraft.getInstance().getResourceManager() == null) return null; // check minecraft for null for CI environments
+        if (!iconSet.isRootIconset) {
```

Comment:
The if shouldn't be necessary because the while loop already checks for the same condition, therefore not running if the condition isn't met at the start.

Reply:
It would always return the current iconset's texture if it couldn't find any, instead of null
even when the current iconset's texture doesn't exist.

Reply:
My solution to the fluid texture issue is to make it check for null when actually adding the textures.

Reply:
The code itself is fine, it just looks like the outer if is redundant, as the `while` loop would check for that anyway before running ^^

Reply:
it doesn't. On fabric, it crashes instead.

Reply:
Oh okay, that's weird...